### PR TITLE
Ensure we spawn the block import worker as an essential task

### PR DESCRIPTION
**Pull request by bkchr**
This pr ensures that we spawn the block import worker as an essential
task. This is quite important as we need to bring down the node when the
block import is done. Besides that it adds some debug output to the
block import worker.

polkadot companion: https://github.com/paritytech/polkadot/pull/2478

### CODE REVIEW

File: primitives/core/src/traits.rs [hunk 1]
```diff
@@ -227,3 +227,28 @@ impl SpawnNamed for Box<dyn SpawnNamed> {
...
+	///
+	/// The given `name` is used to identify the future in tracing.
+	fn spawn_essential_blocking(&self, name: &'static str, future: futures::future::BoxFuture<'static, ()>);
+	/// Spawn the given non-blocking future.
+	///
+	/// The given `name` is used to identify the future in tracing.
+	fn spawn_essential(&self, name: &'static str, future: futures::future::BoxFuture<'static, ()>);
+}
+
+impl SpawnEssentialNamed for Box<dyn SpawnNamed> {
```
Comment (andresilva) [inline primitives/core/src/traits.rs:246]: This is not what you want 🙈

**Review by andresilva — CHANGES_REQUESTED** [review]

File: primitives/core/src/traits.rs (hunk 1 above)
Comment (bkchr) [reply primitives/core/src/traits.rs:246]: 🤦

**Review by andresilva — APPROVED** [review]

File: primitives/core/src/testing.rs [hunk 2]
```diff
@@ -152,3 +152,13 @@ impl crate::traits::SpawnNamed for TaskExecutor {
 		self.0.spawn_ok(future);
 	}
 }
+
+#[cfg(feature = "std")]
+impl crate::traits::SpawnEssentialNamed for TaskExecutor {
```
Comment (kianenigma) [inline primitives/core/src/testing.rs:157]: where is this used? and why is the name not used?

**Review by kianenigma — APPROVED** [review]
A bit outside of my zone of knowledge, but looks sensible.

File: primitives/core/src/testing.rs (hunk 2 above)
Comment (bkchr) [reply primitives/core/src/testing.rs:157]: As the file name indicates, this is only for testing purposes :D

**Comment by bkchr** [comment]
bot merge

**Pull request merged**